aiocache/decorators.py

```diff
@@ -55,7 +55,9 @@ def cached_decorator(func):
 
             try:
                 if await cache_instance.exists(cache_key):
-                    return await cache_instance.get(cache_key)
+                    result = await cache_instance.get(cache_key)
+                    await cache_instance.close()
```

Comment:
Yeah, I'm not talking about this kind of speed but speed for a single call. In order for the function to return, it will have to wait for the pool to close (and the loop to reschedule it) when it really doesn't have to to retrieve the result.

In an API with high load, this can lead to unnecessary delayed calls. Imagine the following case:

1. New call arrives to the decorated function F
2. Other 1000 async calls arrive (no need to be on the same function, any async will do)
3. F gets the loop and executes until arriving to the `await close`. (I know there are more breaks in the middle but for the sake of simplicity lets imagine we are there already). Because of finding the `await` it is rescheduled to the queue of tasks to execute (as the last one because its round robin).
4. Until all 1000 async tasks aren't processed (i.e. find their first break), the loop won't pick again  F and it will pick it just to return something it could have returned already.

Also there can be side effects when trying to close the connection (https://github.com/aio-libs/aioredis/blob/master/aioredis/connection.py#L315) that raise exceptions.

Because all of that I prefer to use `ensure_future`. I agree with you that its a code smell and we won't be freeing all the resources but I prefer it that way. I have in mind a way to solve these situations, when I mature the idea a bit more I'll ping you to validate it if that's okay.

I hope this convinces you :)